Uniformizing EditDrinkForm and small optimizations

AndrewLiaIH opened the pull request.

AndrewLiaIH committed:
Uniformizing EditDrinkForm and small optimizations

AnastasiaIU commented on SomerenDAL/ActivityDao.cs:
```diff
@@ -12,8 +12,7 @@ public List<Activity> GetAllActivities()
             SqlParameter[] sqlParameters = new SqlParameter[0];
-            DataTable dataTable = ExecuteSelectQuery(query, sqlParameters);
-            return ReadTables(dataTable);
+            return ReadTables(ExecuteSelectQuery(query, sqlParameters));
```
I would like to keep the two-line approach for the following reasons:
1. Debugging: It's easier to inspect the value of dataTable in a debugger when it's assigned to a variable on its own line.
2. If you need to add error handling of the dataTable before passing it to ReadTables, having it as a separate line makes this insertion simpler.
3. This can be slightly harder to read, especially for complex method calls or when chaining multiple methods together, as it packs more logic into a single line.

AnastasiaIU commented on SomerenUI/EditDrinkForm.cs:
```diff
@@ -31,40 +29,38 @@ private void btnCreateDrink_Click(object sender, EventArgs e)
+                if (txtDrinkName.Text.IsNullOrEmpty())                
+                    throw new ArgumentNullException();                
 
                 Drink drink = new Drink(double.Parse(txtDrinkPrice.Text), isAlcoholic, txtDrinkName.Text, int.Parse(txtDrinkStock.Text));
```
I suggest breaking down operations into more digestible steps here.
Splitting operations is a widely accepted coding practice, especially regarding readability and maintainability. Breaking down operations into more digestible steps, particularly when it involves parsing or converting data types, can significantly enhance clarity and ease of debugging.

AnastasiaIU commented on SomerenUI/Properties/Resources.resx:
```diff
@@ -146,6 +146,10 @@ Error: </value>
   <data name="No" xml:space="preserve">
     <value>No</value>
   </data>
+  <data name="NullField" xml:space="preserve">
```
I suggest renaming the resource to "ErrorMessageNullField" for consistency.

AnastasiaIU replied:
Sorry, also specify what the error message is about the name, please. Something like: "ErrorMessageNullNameField".

AnastasiaIU commented on SomerenUI/EditDrinkForm.cs:
```diff
@@ -31,40 +29,38 @@ private void btnCreateDrink_Click(object sender, EventArgs e)
+                ShowError(errorMessage, ex);
             }
-        }
+        }        
```
We can remove unnecessary spaces here.

AnastasiaIU commented on SomerenUI/Properties/Resources.resx:
```diff
@@ -162,6 +166,9 @@ Error: </value>
     <value>Successfully deleted: </value>
   </data>
+  <data name="SuccessfullyEdited" xml:space="preserve">
+    <value>Successfully edited:</value>
```
Need to add a space after ":".

AnastasiaIU reviewed (changes requested):
Please, address this small feedback before the merge. Thank you for your improvements and review!

AndrewLiaIH committed:
Requested changes + read commit message

Made suggested changes from pull review. Added an form.Close option in ShowMessage, since all instances besides one had Close right after calling ShowMessage. Corrected TabIndex of pop-up forms. Changed button sizing to not autosize and always be 30 units apart

AnastasiaIU commented on SomerenUI/BaseForm.cs:
```diff
@@ -72,9 +72,11 @@ protected void ShowError(string errorMessage, Exception ex)
         }
 
-        protected void ShowMessage(string message, string arg = null)
+        protected void ShowMessage(string message, string arg = null, Form form = null)
```
It breaks the single responsibility rule and reusability of this method. We could rename it to ShowMessageAndCloseForm but this method is a common case and we might use it further without the closing part. So, I would keep it as it is.

AnastasiaIU commented on SomerenUI/EditDrinkForm.cs:
```diff
@@ -29,20 +29,22 @@ private void btnCreateDrink_Click(object sender, EventArgs e)
             catch (Exception ex)
             {
-                string errorMessage = ex is ArgumentNullException ? Properties.Resources.NullField : Properties.Resources.ErrorMessage;
+                string errorMessage = ex is ArgumentNullException ? Properties.Resources.ErrorMessageNullField : Properties.Resources.ErrorMessage;
```
The error message can be used only for an empty Name field: "Need to input a name. Error: " but the naming "ErrorMessageNullField" suggests that I can use it for any NullField errors. Please, make it clearer like "ErrorMessageNullNameField".

AnastasiaIU reviewed (changes requested):
1. Change queries to contain ORDER BY.
2. Change the error message for NullField.
3. Add a separate methid ShowMessageAndCloseForm.
4. Handle dots and commas for the price.

AndrewLiaIH committed:
Requested changes + read commit message

Dao OrderBy, ShowMessageAndCloseForm, Invariant culture for money, radio button in constructor

AnastasiaIU reviewed (approved):
Approved. Need to remove usage of the currentDrink in btnCreateDrink_Click.
